pkg/chunk/aws/s3_storage_client.go

```diff
@@ -231,8 +239,55 @@ func buildS3Config(cfg S3Config) (*aws.Config, []string, error) {
 		return nil, nil, errors.New("must supply both an Access Key ID and Secret Access Key or neither")
 	}
 
+	role := os.Getenv("AWS_ROLE_ARN")
```

Comment:
The default credential resolver should support web identity already; I don't think we need to implement assume role with web identity manually here.